File: src/main/java/team4/footwithme/vote/api/request/annotation/validator/DuplicateValidator.java
package team4.footwithme.vote.api.request.annotation.validator;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;
import team4.footwithme.vote.api.request.annotation.Duplicate;

import java.util.List;

public class DuplicateValidator implements ConstraintValidator<Duplicate, List<?>> {
    @Override
    public boolean isValid(List<?> list, ConstraintValidatorContext context) {
        return list.stream().distinct().count() == list.size();
    }
}

File: src/main/java/team4/footwithme/vote/api/request/annotation/validator/DuplicateValidator.kt
package team4.footwithme.vote.api.request.annotation.validator

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import team4.footwithme.vote.api.request.annotation.Duplicate

class DuplicateValidator : ConstraintValidator<Duplicate?, List<*>> {
    override fun isValid(list: List<*>, context: ConstraintValidatorContext): Boolean {
        return list.stream().distinct().count() == list.size.toLong()
    }
}
